### DISCUSSION TITLE
Light shaft is supported?

### DISCUSSION THREAD

**Discussion by ohmaya** [o3de/o3de#14072] [General]
https://development--o3deorg.netlify.app/docs/user-guide/visualization/environments/vegetation/
https://development--o3deorg.netlify.app/images/shared/dynamic-vegetation-intro.png

Looks like there is light shaft in the picture.

**Comment by AMZN-daimini** [comment]
Hey there! I have investigated a similar question before and it seems life volumetric fog is implemented, but could not find any reference to light shafts.
@o3de/sig-graphics-audio for more info

**Reply by ohmaya** [reply]
So, how do I use volumetric.*fog (Can't find it in the source code)?

**Reply by AMZN-daimini** [reply]
This is what I was directed to at the time
https://www.o3de.org/docs/user-guide/components/reference/atom/deferred-fog/

**Reply by ohmaya** [reply]
That does not seem to be able to do anything similar to https://docs.unrealengine.com/5.1/en-US/volumetric-fog-in-unreal-engine/

**Reply by invertednormal** [reply]
Volumetric fog is a very basic fog type that can't do anything like this. While O3de doesn't currently support light shafts on all light types, you can get sun shafts using the sky atmosphere component. Unfortunately it doesn't have documentation written yet afaik.

**Reply by ohmaya** [reply]
@invertednormal Then how Sky Atmosphere can be used to achieve https://development--o3deorg.netlify.app/images/shared/dynamic-vegetation-intro.png?

**Reply by AMZN-alexpete** [reply]
The sky atmosphere volumetric light cannot yet achieve the smoothness of the light shafts in that picture - IIRC, those light shafts in the picture you posted were based on an older technique that was not volumetric (full screen pass that walks from pixel to sun light source), but LY did have volumetric fog that could do something similar, but I don't think that's what you see in that particular screenshot.

With sky atmosphere you can turn on shadows in the advanced section of the component and turn off fast sky, but even with the samples way up (which makes performance bad) you won't get very smooth light shafts, and that is because you really want to enable TAA to smooth out the samples, so you can do fewer noisy samples but over a few frames they get blended and it looks much better and performs better - that is a goal for a future date.

**Reply by ohmaya** [reply]
Thanks, @AMZN-alexpete
I'm able to see the effect of Sky Atmosphere, but I don't see Enable shadows (Fast sky off) + an opaque entity produces light shaft occlusion effect.

**Reply by AMZN-alexpete** [reply]
I noticed this too recently in my testing and I believe there is a bug in the sky atmosphere and it no longer is using either the correct shader variant (the one that enabled shadows) or it isn't pulling the shadow information from the shadow targets correctly anymore.
